confdgnmi/testtool/OpenConfig/01__interfaces.robot

```diff
@@ -0,0 +1,195 @@
+
+Get "CONFIG" response does not include "state" container
+    [Documentation]    Verify that ``GetRequest`` with ``type=CONFIG`` parameter receives
+    ...                OK response that does not include the read-only YANG "state" container.
+    [Tags]  config  container  negative
+    Set Datatype To    CONFIG
+    Verify Get of  /interfaces/interface[name=${INTERFACE}]
+    Check last updates not include    state
+
+Get "OPERATIONAL" response includes "state" container
+    [Documentation]    Verify that ``GetRequest`` with ``type=OPERATIONAL`` parameter receives
+    ...                OK response that does include the read/write YANG "state" container.
+    [Tags]  operational  container
+    Set Datatype To    OPERATIONAL
+    Verify Get of  /interfaces/interface[name=${INTERFACE}]
+    Check last updates include  state
+
+Get "OPERATIONAL" response does not include "config" container
+    [Documentation]    Verify that ``GetRequest`` with ``type=OPERATIONAL`` parameter receives
+    ...                OK response that does not include the read-only YANG "config" container.
+    [Tags]  operational  container  negative
+    Set Datatype To    OPERATIONAL
+    Verify Get of  /interfaces/interface[name=${INTERFACE}]
+    Check last updates not include  config
+
+Get "ALL" response includes both "config" and "state" containers
+    [Documentation]    Verify that ``GetRequest`` with ``type=ALL`` parameter receives
+    ...                OK response that does include both read-write "config" and read-only "state" containers.
+    [Tags]  config  operational  container
+    Set Datatype To    ALL
+    Verify Get of  /interfaces/interface[name=${INTERFACE}]
+    Check last updates include  config
+    Check last updates include  state
+
+Get "config" response includes significant leaves
+    [Documentation]    Verify that ``GetRequest`` with ``type=CONFIG`` parameter receives
+    ...                OK response that does include standardized leaf names/items.
+    [Tags]  config  leaf
+    [Template]    Get interface "config" includes leaf
+    name
+    type
+    mtu
+    loopback-mode
+    description
+    enabled
+
+Get "state" response includes significant leaves
+    [Documentation]    Verify that ``GetRequest`` with ``type=OPERATIONAL`` parameter receives
+    ...                OK response that does include standardized leaf names/items.
+    [Tags]  operational  leaf
+    # TODO - add all items to verify deviation from full official OC YANG?
+    [Template]    Get interface "state" includes leaf
+    name
+    type
+    mtu
+    enabled
+    ifindex
+    admin-status
+    oper-status
+
+Get "config" works for all supported encodings
+    [Documentation]    Verify that ``GetRequest`` with ``type=CONFIG``
+    ...                and ``path=interfaces/interface[]/config`` parameters receives
+    ...                OK response for all/any of the supported ``encoding` parameter values.
+    [Tags]  config  container  encoding
```

Comment:
IDK, there already is "Get works for all encodings", does this add something substantial?

Reply:
I agree that it smells a bit of "random parameters combination" - i tried to come with possibly typical end-user request/RPC...
Intended "added value" if this test-case was that seemingly core/most important "config true" item declared in standard YANG  is included for "type=CONFIG" request independent of encoding parameter.

Reply:
To answer the initial question -> generic test reaches for root of data tree, while this one specifically for the "CONFIG" node, thus potentially logical difference that could result in other response (assuming server is not ideally implemented).

This test-case definitely falls into "potentially unnecessary" group, but can as well spot one type of bugs in server implementation IMHO.